depthai_nodes/ml/parsers/ppocr_text_detection.py

```diff
@@ -0,0 +1,103 @@
+import depthai as dai
+import numpy as np
+
+from ..messages.creators import create_detection_message
+from .utils.ppocr import corners2xyxy, parse_paddle_detection_outputs
+
+
+class PPTextDetectionParser(dai.node.ThreadedHostNode):
+    """Parser class for parsing the output of the PP-OCR text detection model.
+
+    Attributes
+    ----------
+    input : Node.Input
+        Node's input. It is a linking point to which the Neural Network's output is linked. It accepts the output of the Neural Network node.
+    out : Node.Output
+        Parser sends the processed network results to this output in a form of DepthAI message. It is a linking point from which the processed network results are retrieved.
+    mask_threshold : float = 0.3
+        The threshold for the mask.
+    bbox_threshold : float = 0.7
+        The threshold for bounding boxes.
+    max_detections : int = 1000
+        The maximum number of candidate bounding boxes.
+
+    Output Message/s
+    -------
+    **Type**: dai.ImgDetections
+    **Description**: ImgDetections message containing bounding boxes and the respective confidence scores of detected text.
+    """
+
+    def __init__(
+        self,
+        mask_threshold: float = 0.3,
+        bbox_threshold: float = 0.7,
+        max_detections: int = 1000,
+    ):
+        """Initializes the PPTextDetectionParser node.
+
+        @param mask_threshold: The threshold for the mask.
+        @type mask_threshold: float = 0.3
+        @param bbox_threshold: The threshold for bounding boxes.
+        @type bbox_threshold: float = 0.7
+        @param max_detections: The maximum number of candidate bounding boxes.
+        @type max_detections: int = 1000
+        """
+        dai.node.ThreadedHostNode.__init__(self)
+        self.input = self.createInput()
+        self.out = self.createOutput()
+
+        self.mask_threshold = mask_threshold
+        self.bbox_threshold = bbox_threshold
+        self.max_detections = max_detections
+
+    def setMaskThreshold(self, threshold: float = 0.3):
+        """Sets the mask threshold for creating the mask from model output
+        probabilities.
+
+        @param threshold: The threshold for the mask.
+        @type threshold: float
+        """
+        self.conf_threshold = threshold
```

Comment:
`conf_threshold` is not defined. Did you mean `mask_threshold`? Does `mask_threshold` here have the same meaning as `conf_threshold`? If yes, please rename it to `conf_threshold` for consistency.

Reply:
No, for detection decoding it has to be seperate as its a different model with completely different outputs.

It should be mask_threshold, thanks for noticing. conf_threshold was first then I changed it to mask_threshold.